tests/cases/exporting/tests.py

```diff
@@ -104,106 +152,155 @@ def setUp(self):
             'title__salary')
 
     def test_csv(self):
+        name = 'export.csv'
+        exp_size = 250
+
         exporter = export.CSVExporter(self.concepts)
-        buff = exporter.write(self.query)
-        buff.seek(0)
-        self.assertEqual(len(buff.read()), 246)
+        it = exporter.read(self.query)
+
+        exporter.write(it, buff=name)
+
+        size = os.path.getsize(name)
+        self.assertAlmostEqual(size, exp_size, delta=delta)
+
+        os.remove(name)
 
     def test_excel(self):
-        fname = 'excel_export.xlsx'
+        name = 'excel_export.xlsx'
+        exp_size = 6120
+
         exporter = export.ExcelExporter(self.concepts)
-        exporter.write(self.query, fname)
-        self.assertTrue(os.path.exists(fname))
-        # Observed slight size differences..
-        l = len(open(fname).read())
-        self.assertTrue(6170 <= l <= 6250)
-        os.remove(fname)
+        it = exporter.read(self.query)
+        exporter.write(it, buff=name)
+
+        size = os.path.getsize(name)
+        self.assertAlmostEqual(size, exp_size, delta=delta)
+
+        os.remove(name)
 
     def test_sas(self):
-        fname = 'sas_export.zip'
+        name = 'sas_export.zip'
+        exp_size = 1340
+
         exporter = export.SASExporter(self.concepts)
-        exporter.write(self.query, fname)
-        self.assertTrue(os.path.exists(fname))
-        self.assertEqual(len(open(fname).read()), 1335)
-        os.remove(fname)
+        it = exporter.read(self.query)
+        exporter.write(it, buff=name)
+
+        size = os.path.getsize(name)
+        self.assertAlmostEqual(size, exp_size, delta=delta)
+
+        os.remove(name)
 
     def test_r(self):
-        fname = 'r_export.zip'
+        name = 'r_export.zip'
+        exp_size = 760
+
         exporter = export.RExporter(self.concepts)
-        exporter.write(self.query, fname)
-        self.assertTrue(os.path.exists(fname))
-        self.assertEqual(len(open(fname).read()), 754)
-        os.remove(fname)
+        it = exporter.read(self.query)
+        exporter.write(it, buff=name)
+
+        size = os.path.getsize(name)
+        self.assertAlmostEqual(size, exp_size, delta=delta)
+
+        os.remove(name)
 
     def test_json(self):
+        name = 'export.json'
+        exp_size = 630
+
         exporter = export.JSONExporter(self.concepts)
-        buff = exporter.write(self.query)
-        buff.seek(0)
-        self.assertEqual(len(buff.read()), 639)
+        it = exporter.read(self.query)
+
+        exporter.write(it, buff=name)
+
+        size = os.path.getsize(name)
+        self.assertAlmostEqual(size, exp_size, delta=delta)
+
+        os.remove(name)
 
     def test_html(self):
+        name = 'export.html'
+        exp_size = 1960
```

Comment:
An initial run of the tests with manual inspection of the content. The sizes were then recorded. This is an simpler alternative to re-reading the data and validating it (and admittedly much less fragile).